File: eBarberShop/eBarberShop.Model/NarudzbeDetalji.cs
namespace eBarberShop.Model
{
    public class NarudzbeDetalji
    {
        public int NarudzbeDetaljiId { get; set; }
        public int Kolicina { get; set; }
        public int NarudzbaId { get; set; }
        public int ProizvodId { get; set; }

        public virtual Proizvodi Proizvod { get; set; }

        public decimal? CijenaProizvoda => Proizvod?.Cijena;
        public string? NazivProizvoda => Proizvod?.Naziv;
        public string? SifraProizvoda => Proizvod?.Sifra;

    }
}

File: eBarberShop/UI/ebarbershop_mobile/lib/models/narudzbe/narudzbe_detalji.dart
import 'package:json_annotation/json_annotation.dart';

part 'narudzbe_detalji.g.dart';

@JsonSerializable()
class NarudzbeDetalji {
  int narudzbeDetaljiId;
  int kolicina;
  int narudzbaId;
  int proizvodId;
  double? cijenaProizvoda;
  String? nazivProizvoda;
  String? sifraProizvoda;

  NarudzbeDetalji(
      this.narudzbeDetaljiId,
      this.kolicina,
      this.narudzbaId,
      this.proizvodId,
      this.cijenaProizvoda,
      this.nazivProizvoda,
      this.sifraProizvoda);

  factory NarudzbeDetalji.fromJson(Map<String, dynamic> json) =>
      _$NarudzbeDetaljiFromJson(json);

  Map<String, dynamic> toJson() => _$NarudzbeDetaljiToJson(this);
}
